### ISSUE TITLE
Incorrect rule successfully typechecking

### ISSUE THREAD

**Issue by Gaspi** [Deducteam/Dedukti#17]
The following system successfully typechecks in Dedukti:
```
def A : Type.
def eA : A -> Type.
a : A.
[] A --> eA a.

B : Type.
def eB : B -> Type.

[] eB a --> eA a.
```
even though `eB` has type `B -> Type` and `a` has type `eA a` with `eA a` and `B` are not convertible.

Dedukti should only successfully typecheck rules when the following two conditions are met:
1) All substitution yielding a well typed term from the left hand side pattern also yield a well typed term when applied to the right hand side term. 
2) There exists such a substitution (yielding a well typed term from the left hand side pattern)

**Comment by francoisthire** [comment]
```dedukti
#NAME test.

type : Type.
def eta : type -> Type.
A : type.
a : eta A.
B : type.
b : eta B.
or : type -> type -> type.


def injl : A:type -> B:type -> eta A -> eta (or A B).

def injr : A:type -> B:type -> eta B -> eta (or A B).

U : eta (or A B) -> type.

[] eta A --> eta (U (injl A B a)).

[] eta B --> eta (U (injr A B b)).

[] injl B B a --> injr B B b.
```

Another similar example. Probably this one is an instance of the previous one but we never know...

**Comment by rlepigre** [comment]
That's not a bug. The terms that match this pattern are not typable, so the rule cannot be applied. The example is also accepted by my implementation.

**Comment by GuillaumeGen** [comment]
I am no able to find a wrongly typed term using this feature (yet).
But the rule is used, since ```#CHECK a,eB a.``` answers yes in the current implementation of Dedukti, even if eB a is ill-typed.

**Comment by rlepigre** [comment]
Well that's indeed a bug. The problem is that Dedukti (and my prototype as well) do not check that the "type" `eB a` has type `Type` (or `Kind`).

**Comment by francoisthire** [comment]
Fixed in #108 .
